demos/cli_uart/cli_uart_demo.c

```diff
@@ -0,0 +1,297 @@
+ * parameter is echoes back one at a time.
+ */
+static BaseType_t prvParameterEchoCommand( char * pcWriteBuffer,
+                                           size_t xWriteBufferLen,
+                                           const char * pcCommandString );
+
+/*
+ * Defines a command that returns a table showing the state of each task at the
+ * time the command is called.
+ */
+static BaseType_t prvTaskStatsCommand( char * pcWriteBuffer,
+                                       size_t xWriteBufferLen,
+                                       const char * pcCommandString );
+
+
+/* Structure that defines the "echo_3_parameters" command line command.  This
+ * takes exactly three parameters that the command simply echos back one at a
+ * time. */
+static const CLI_Command_Definition_t xThreeParameterEcho =
+{
+    "echo_3_parameters",
+    "\r\necho_3_parameters <param1> <param2> <param3>:\r\n Expects three parameters, echos each in turn\r\n\r\n",
+    prvThreeParameterEchoCommand, /* The function to run. */
+    3                             /* Three parameters are expected, which can take any value. */
+};
+
+/* Structure that defines the "echo_parameters" command line command.  This
+ * takes a variable number of parameters that the command simply echos back one at
+ * a time. */
+static const CLI_Command_Definition_t xParameterEcho =
+{
+    "echo_parameters",
+    "\r\necho_parameters <...>:\r\n Take variable number of parameters, echos each in turn\r\n\r\n",
+    prvParameterEchoCommand, /* The function to run. */
+    -1                       /* The user can enter any number of commands. */
+};
+
+/* Structure that defines the "task-stats" command line command. */
+static const CLI_Command_Definition_t xTaskStats =
+{
+    "task-stats",        /* The command string to type. */
+    "\r\ntask-stats:\r\n Displays a table showing the state of each FreeRTOS task\r\n\r\n",
+    prvTaskStatsCommand, /* The function to run. */
+    0                    /* No parameters are expected. */
+};
+
+/**
+ * Static array into which the commands will be placed from the console.
+ */
+static char cCommandBuffer[ cmdMAX_INPUT_SIZE ];
+
+/**
+ * Static array into which the output of the command will be stored to be written to the console.
+ */
+static char cOutputBuffer[ cmdMAX_OUTPUT_SIZE ];
+
+static BaseType_t prvThreeParameterEchoCommand( char * pcWriteBuffer,
+                                                size_t xWriteBufferLen,
+                                                const char * pcCommandString )
+{
+    const char * pcParameter;
+    BaseType_t lParameterStringLength, xReturn;
+    static BaseType_t lParameterNumber = 0;
+
+    /* Remove compile time warnings about unused parameters, and check the
+     * write buffer is not NULL.  NOTE - for simplicity, this example assumes the
+     * write buffer length is adequate, so does not check for buffer overflows. */
+    ( void ) pcCommandString;
+    ( void ) xWriteBufferLen;
```

Comment:
These parameters are being used, is this still necessary to remove the warnings?

Reply:
I think you are right. The parameter `xWriteBufferLen` is definitely used though `pcCommandString`  is just passed onto other functions. I have removed those statements.